overlord/hookstate/ctlcmd/health.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/healthstate"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+var (
+	shortHealthHelp = i18n.G("Report on snap's health")
+	longHealthHelp  = i18n.G(`
+The set-health command is called from a snap to inform the system of the snap's
+overall health.
+
+Note the health is of the snap, not of the apps it contains; it’s up to the
+snap developer to determine how the health of the individual apps add up to
+the health of the snap as a whole.
```

Comment:
didn't we discuss mentioning check-health explicitly here though?

Reply:
I don't remember that, but maybe we did. Can you point me at it?

Reply:
this comment mentions it:  https://github.com/snapcore/snapd/pull/6848#discussion_r288614070

> Also this should probably mention the hook as a common usage place.

Reply:
How about this, just before the Note:
> It can be called from any hook, and even from the apps themselves,
although a snap can provide a 'check-health' hook that is solely
concerned with this; if it does, that hook is called periodically
(with increased frequency while the snap is "unhealthy"). Any health
regression issues a warning to the user.

the last line would be untrue until we implemented that (I could add it then)